R/config.R

```diff
@@ -243,27 +260,66 @@ validate_config_targets <- function(predevals_config, task_groups, task_id_names
 }
 
 
-#' Validate the eval_windows in a predevals config object
-#'  - check that min_round_id specified in predevals config is a valid round_id
+#' Validate the eval_sets in a predevals config object
+#'  - check that min specified in predevals config is a valid round_id
 #'    for the hub
+#'  - check that any entries in task_filters are valid task id variables
+#'  - check that for each task id variable in task_filters, specified values are valid values
+#'    for that task id as specified in the hub's config
 #'
 #' @noRd
-validate_config_eval_windows <- function(predevals_config, hub_tasks_config) {
+validate_config_eval_sets <- function(predevals_config, hub_tasks_config, task_groups, task_id_names) {
   hub_round_ids <- hubUtils::get_round_ids(hub_tasks_config)
-  for (eval_window in predevals_config$eval_windows) {
-    # check that min_round_id is a valid round_id
-    # only do this check if eval_window$min_round_id is specified
-    if (!"min_round_id" %in% names(eval_window)) {
-      next
+  for (eval_set in predevals_config$eval_sets) {
+    # check that min is a valid round_id
+    # only do this check if eval_set$min is specified
+    round_filters <- eval_set$round_filters
+    if ("min" %in% names(round_filters) && !round_filters$min %in% hub_round_ids) {
+      raise_config_error(
+        cli::format_inline(
+          "Minimum round id {.val {round_filters$min}} for evaluation ",
+          "set is not a valid round id for the hub."
+        )
+      )
     }
-    if (!eval_window$min_round_id %in% hub_round_ids) {
+
+    # check that any entries in task_filters are valid task id variables
+    task_filters <- eval_set$task_filters
+    extra_set_filter_names <- setdiff(
+      names(task_filters),
+      task_id_names
+    )
+    if (length(extra_set_filter_names) > 0) {
       raise_config_error(
         cli::format_inline(
-          "Minimum round id {.val {eval_window$min_round_id}} for evaluation ",
-          "window is not a valid round id for the hub."
```

Comment:
This diff is not useful -- the minimum round id message has not been deleted, it's on line 280 above.